tests/unit_tests/conditional/ci/test_categorical_test.py

```diff
@@ -0,0 +1,156 @@
+        x_vars={"Education"},
+        y_vars={"HoursPerWeek"},
+        z_covariates=["Age", "Immigrant", "Race", "Sex"],
+        df=df_adult,
+    )
+    assert_almost_equal(coef, 1460.11, decimal=1)
+    assert_almost_equal(p_value, 0, decimal=1)
+    assert ci_est.dof_ == 316
+
+    coef, p_value = ci_est.test(x_vars={"Immigrant"}, y_vars={"Sex"}, z_covariates={}, df=df_adult)
+    assert_almost_equal(coef, 0.2724, decimal=1)
+    assert_almost_equal(np.log(p_value), -0.50, decimal=1)
+    assert ci_est.dof_ == 1
+
+    coef, p_value = ci_est.test(
+        x_vars={"Education"}, y_vars={"MaritalStatus"}, z_covariates=["Age", "Sex"], df=df_adult
+    )
+    assert_almost_equal(coef, 481.96, decimal=1)
+    assert_almost_equal(p_value, 0, decimal=1)
+    assert ci_est.dof_ == 58
+
+    # Values differ (for next 2 tests) from dagitty because dagitty ignores grouped
+    # dataframes with very few samples. Update: Might be same from scipy_vars=1.7.0
+    coef, p_value = ci_est.test(
+        x_vars={"Income"},
+        y_vars={"Race"},
+        z_covariates=["Age", "Education", "HoursPerWeek", "MaritalStatus"],
+        df=df_adult,
+    )
+
+    assert_almost_equal(coef, 66.39, decimal=1)
+    assert_almost_equal(p_value, 0.99, decimal=1)
+    assert ci_est.dof_ == 136
+
+    coef, p_value = ci_est.test(
+        x_vars={"Immigrant"},
+        y_vars={"Income"},
+        z_covariates=["Age", "Education", "HoursPerWeek", "MaritalStatus"],
+        df=df_adult,
+    )
+    assert_almost_equal(coef, 65.59, decimal=1)
+    assert_almost_equal(p_value, 0.999, decimal=2)
+    assert ci_est.dof_ == 131
+
+
+def test_discrete_tests():
+    for t in [
+        CategoricalCITest("pearson"),  # chi-square
+        CategoricalCITest("log-likelihood"),  # G^2
+        CategoricalCITest("freeman-tukey"),  # freeman-tukey
+        CategoricalCITest("mod-log-likelihood"),  # Modified log-likelihood
+        CategoricalCITest("neyman"),  # Neyman
+        CategoricalCITest("cressie-read"),  # Cressie-read
+    ]:
+        assert (
+            t.test(
+                x_vars={"Age"},
+                y_vars={"Immigrant"},
+                z_covariates=[],
+                df=df_adult,
+            )[1]
+            < 0.05
+        )
+
+        assert (
+            t.test(
+                x_vars={"Age"},
+                y_vars={"Race"},
+                z_covariates=[],
+                df=df_adult,
+            )[1]
+            < 0.05
+        )
+
+        assert (
+            t.test(
+                x_vars={"Age"},
+                y_vars={"Sex"},
+                z_covariates=[],
+                df=df_adult,
+            )[1]
+            < 0.05
+        )
+        assert not (
+            t.test(
+                x_vars={"Immigrant"},
+                y_vars={"Sex"},
+                z_covariates=[],
+                df=df_adult,
+            )[1]
+            < 0.05
+        )
+
+        # XXX: Test returns nan...
```

Comment:
Remove these commented lines. Also, if the test returns nan, make sure you understand why to avoid that there is actually a bug.

Reply:
Done